diff --git a/config/pexos/waagent.conf b/config/pexos/waagent.conf
new file mode 100644
index 0000000..d4f2787
--- /dev/null
+++ b/config/pexos/waagent.conf
@@ -0,0 +1,76 @@
+#
+# Microsoft Azure Linux Agent Configuration
+#
+
+# Specified program is invoked with the argument "Ready" when we report ready status
+# to the endpoint server.
+Role.StateConsumer=None
+
+# Specified program is invoked with XML file argument specifying role
+#  configuration.
+Role.ConfigurationConsumer=None
+
+# Specified program is invoked with XML file argument specifying role topology.
+Role.TopologyConsumer=None
+
+# Enable instance creation
+Provisioning.Enabled=n
+
+# Password authentication for root account will be unavailable.
+Provisioning.DeleteRootPassword=y
+
+# Generate fresh host key pair.
+Provisioning.RegenerateSshHostKeyPair=n
+
+# Supported values are "rsa", "dsa" and "ecdsa".
+Provisioning.SshHostKeyPairType=rsa
+
+# Monitor host name changes and publish changes via DHCP requests.
+Provisioning.MonitorHostName=n
+
+# Decode CustomData from Base64.
+Provisioning.DecodeCustomData=n
+
+# Execute CustomData after provisioning.
+Provisioning.ExecuteCustomData=n
+
+# Algorithm used by crypt when generating password hash.
+#Provisioning.PasswordCryptId=6
+
+# Length of random salt used when generating password hash.
+#Provisioning.PasswordCryptSaltLength=10
+
+# Format if unformatted. If 'n', resource disk will not be mounted.
+ResourceDisk.Format=n
+
+# File system on the resource disk
+# Typically ext3 or ext4. FreeBSD images should use 'ufs2' here.
+ResourceDisk.Filesystem=ext4
+
+# Mount point for the resource disk
+ResourceDisk.MountPoint=/mnt
+
+# Create and use swapfile on resource disk.
+ResourceDisk.EnableSwap=n
+
+# Size of the swapfile.
+ResourceDisk.SwapSizeMB=0
+
+# Respond to load balancer probes if requested by Microsoft Azure.
+LBProbeResponder=n
+
+# Enable verbose logging (y|n)
+Logs.Verbose=n
+
+# Root device timeout in seconds.
+OS.RootDeviceScsiTimeout=300
+
+# If "None", the system default version is used.
+OS.OpensslPath=None
+
+# If set, agent will use proxy server to access internet
+#HttpProxy.Host=None
+#HttpProxy.Port=None
+
+# Detect Scvmm environment, default is n
+# DetectScvmmEnv=n
diff --git a/init/pexos/walinuxagent b/init/pexos/walinuxagent
new file mode 100644
index 0000000..af5ec3b
--- /dev/null
+++ b/init/pexos/walinuxagent
@@ -0,0 +1,2 @@
+# To enable the Microsoft Azure Agent, set WALINUXAGENT_ENABLED=1
+WALINUXAGENT_ENABLED=0
diff --git a/setup.py b/setup.py
index 0f6230e..3533f9f 100755
--- a/setup.py
+++ b/setup.py
@@ -105,6 +105,12 @@ def get_data_files(name, version, fullname):
         else:
             #sles 12+ and openSUSE 13.2+ use systemd
             set_systemd_files(data_files, dest='/usr/lib/systemd/system')
+    elif name == 'pexos':
+        set_bin_files(data_files)
+        set_conf_files(data_files, src=["config/pexos/waagent.conf"])
+        set_logrotate_files(data_files)
+        set_files(data_files, dest="/etc/default",
+                  src=["init/pexos/walinuxagent"])
     else:
         #Use default setting
         set_bin_files(data_files)
diff --git a/azurelinuxagent/distro/pexos/__init__.py b/azurelinuxagent/distro/pexos/__init__.py
new file mode 100644
index 0000000..e69de29
--- /dev/null
+++ b/azurelinuxagent/distro/pexos/__init__.py
@@ -0,0 +1,19 @@
+# Microsoft Azure Linux Agent
+#
+# Copyright 2014 Microsoft Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Requires Python 2.4+ and Openssl 1.0+
+#
+
diff --git a/azurelinuxagent/distro/pexos/loader.py b/azurelinuxagent/distro/pexos/loader.py
new file mode 100644
index 0000000..12b0428
--- /dev/null
+++ b/azurelinuxagent/distro/pexos/loader.py
@@ -0,0 +1,3 @@
+def get_osutil():
+    from azurelinuxagent.distro.pexos.osutil import PexOSUtil
+    return PexOSUtil()
diff --git a/azurelinuxagent/distro/pexos/osutil.py b/azurelinuxagent/distro/pexos/osutil.py
new file mode 100644
index 0000000..3ec1b17
--- /dev/null
+++ b/azurelinuxagent/distro/pexos/osutil.py
@@ -0,0 +1,7 @@
+from azurelinuxagent.distro.default.osutil import DefaultOSUtil
+
+class PexOSUtil(DefaultOSUtil):
+    def __init__(self):
+        super(PexOSUtil, self).__init__()
+        self.agent_pid_file_path = "/var/run/walinuxagent/waagent.pid"
+
