diff --git a/azurelinuxagent/protocol/protocolFactory.py b/azurelinuxagent/protocol/protocolFactory.py
index 0bf6e52..d0ae532 100644
--- a/azurelinuxagent/protocol/protocolFactory.py
+++ b/azurelinuxagent/protocol/protocolFactory.py
@@ -19,6 +19,7 @@
 import os
 import traceback
 import threading
+import time
 import azurelinuxagent.logger as logger
 from azurelinuxagent.future import text
 import azurelinuxagent.utils.fileutil as fileutil
@@ -57,16 +58,19 @@ def detect_metadata_protocol():
     logger.info("Protocol V2 found.")
     return protocol
 
-def detect_available_protocols(prob_funcs=[detect_wire_protocol, 
-                                           detect_metadata_protocol]):
-    available_protocols = []
-    for probe_func in prob_funcs:
-        try:
-            protocol = probe_func()
-            available_protocols.append(protocol)
-        except ProtocolNotFound as e:
-            logger.info(text(e))
-    return available_protocols
+def detect_available_protocols(prob_funcs=[detect_wire_protocol]):
+    for retry in xrange(0, 360):
+        for probe_func in prob_funcs:
+            try:
+                return [probe_func()]
+            except ProtocolError as e:
+                logger.info(text(e))
+            except ProtocolNotFound as e:
+                logger.info(text(e))
+                raise e
+        if retry < 360-1:
+            time.sleep(10)
+    raise ProtocolNotFound("No protocol found")
 
 def detect_default_protocol():
     logger.info("Detect default protocol.")
@@ -86,7 +90,7 @@ def get_wire_protocol():
 def get_metadata_protocol():
     return MetadataProtocol()
 
-def get_available_protocols(getters=[get_wire_protocol, get_metadata_protocol]):
+def get_available_protocols(getters=[get_wire_protocol]):
     available_protocols = []
     for getter in getters:
         try:
